internal/globaldefault/default.go

```diff
@@ -70,9 +71,16 @@ func SetupDefaultActivation(subshell subshell.SubShell, cfg DefaultConfigurer, r
 		return locale.WrapError(err, "err_globaldefault_rtexes", "Could not retrieve runtime executables")
 	}
 
+	env, err := runtime.Env(false, false)
+	if err != nil {
+		return locale.WrapError(err, "err_globaldefault_rtenv", "Could not construct runtime environment variables")
+	}
+
+	sockPath := svcctl.NewIPCSockPathFromGlobals().String()
```

Comment:
Doesn't seem there's any value in passing this as an arg.

Reply:
"globaldefault" makes use of the more primitive packages like runtime. It seems awkward for the runtime package to have a dependency on the svcctl package when it can, instead, be structured here as a parameter. Also, it makes future refactoring or updating easier. For example, in the case that the NewIPCSockPathFromGlobals function ends up needing some argument.